app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -0,0 +1,289 @@
+package com.github.se.cyrcle.ui.gambling
+
+import android.util.Log
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.graphics.drawscope.rotate
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.drawText
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.rememberTextMeasurer
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.cyrcle.R
+import kotlin.math.*
+import kotlinx.coroutines.*
+
+@Composable
+fun WheelView(
+    modifier: Modifier = Modifier,
+    onSegmentLanded: ((String) -> Unit)? = null
+): () -> Unit {
+  data class Segment(val name: String, val color: Color, val probability: Float)
+
+  val rarityNothing = stringResource(R.string.rarity_nothing)
+  val rarityLegendary = stringResource(R.string.rarity_legendary)
+  val rarityCommon = stringResource(R.string.rarity_common)
+  val rarityRare = stringResource(R.string.rarity_rare)
+  val rarityEpic = stringResource(R.string.rarity_epic)
+
+  val segments = remember {
```

Comment:
This can be  done  in  a matter of seconds  so  we'll  see once  we decide  what the prizes actually  are. In the meantime i like to have a little fun. (Still  havent got a legendary even though I tested the wheel a bunch haha)